src/pdm/backend/base.py

```diff
@@ -7,7 +7,16 @@
 import warnings
 from fnmatch import fnmatch
 from pathlib import Path
-from typing import TYPE_CHECKING, Any, Iterable, Mapping, TypeVar, cast
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Iterable,
+    Mapping,
+    TypeVar,
+    cast,
+)
+
+from typing_extensions import Literal
```

Comment:
We don't declare `typing_extensions` as an explicit dependency, and it isn't worth a new dependency either.

So I'd like to halt this PR, feel free to close it.

Reply:
any plans to drop support for 3.7? it's been EOL since june last year

Reply:
We can do it in the next two minor release(2.3.0), since I prefer the dropping to happen at a release with few other new features.